### PR TITLE
Remove APIs deprecated on WordPress 5.4

### PR DESCRIPTION
**Pull request by youknowriad**
- getReferenceByDistinctEdits All usage are plugins that bundle their own versions of this function and Gutenberg.
 -  PreserveScrollInReorder All usage are plugins that bundle their own versions of Gutenberg.
 - dropZoneUIOnly I think is still used in 3/4 plugins but it seems low impact to remove it.
 - isDismissable I didn't check whether it's used but it's fairly low impact.
 - wp.data.plugins.control still used in a dozen of plugins but I managed to avoid any breakage even if they keep their usage.

There are some APIs deprecated on WP 5.4 that I didn't remove but instead set for removal on WordPress 6.2. These are more impactful than the removed ones:

 - Props on the RichText component.
 - The IconButton component.
 - isDefault prop on the Button component.
 - the wp.nux package. (I hesitated about this one, we could have just removed it maybe now)

### CODE REVIEW

File: packages/data/src/registry.js [hunk 1]
```diff
@@ -291,6 +291,10 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 	// This function will be deprecated as soon as it is no longer internally referenced.
 	//
 	function use( plugin, options ) {
+		if ( ! plugin ) {
+			return;
+		}
```
Comment (Mamaduka) [inline packages/data/src/registry.js:296]: I assume this is a fix for `wp.data.plugins.control`.

**Review by ciampo — COMMENTED** [review]
`Modal`'s `isDismissable` removal LGTM 👍  Thank you!

**Review by Mamaduka — APPROVED** [review]
Changes look good to me, and good call on punting more impactful APIs 👍

File: packages/data/src/registry.js (hunk 1 above)
Comment (youknowriad) [reply packages/data/src/registry.js:296]: yep

**Pull request merged**